GeneralsMD/Code/GameEngine/Source/GameClient/InGameUI.cpp

```diff
@@ -5596,42 +5596,37 @@ void InGameUI::selectNextIdleWorker( void )
 	if(getSelectCount() == 0 || getSelectCount() > 1)
 	{
 		selectThisObject = *m_idleWorkers[index].begin();
+		// If our idle worker is contained by anything, we need to select the container instead.
+		while (selectThisObject->getContainedBy())
+			selectThisObject = selectThisObject->getContainedBy();
 	}
 	else
 	{
 		Drawable *selectedDrawable = TheInGameUI->getFirstSelectedDrawable();	
-		
-		ObjectListIt it = m_idleWorkers[index].begin();
-		while(it != m_idleWorkers[index].end())
+		std::vector<Object*> uniqueIdleWorkers = getUniqueIdleWorkers(m_idleWorkers[index]);
```

Comment:
Better add a `TheSuperHackers @tweak` comment here that explains the change and the user facing benefit it has.